src/util.cc

```diff
@@ -474,35 +478,36 @@ std::string WideToUtf8(const WStringPiece& w_string) {
   int u8_size_needed = WideCharToMultiByte(
       CP_UTF8, 0, w_string.wstr_, static_cast<int>(w_string.size()),
       &u8_string[0], size_needed, NULL, NULL);
-  if (u8_size_needed != size_needed)
-    Fatal("Failed conversion to UTF-8 string");
+  if (u8_size_needed != size_needed) {
+    Win32Fatal("WideToUtf8", "Failed conversion to UTF-8 string");
+  }
 
   return u8_string;
 }
 
-void ConvertWideToUtf8(const WStringPiece& w_string, int size_needed,
-                       char* buff, size_t buffer_length) {
-  if (size_needed > (buffer_length - 1)) // Also account for \0
-    Fatal("Buffer is to small for UTF-8 string");
-
+// Method calling this funcion must ensure that the buffer is large enough to
+// fit the converted string.
+int ConvertWideToUtf8(const WStringPiece& w_string, char* buff,
+                      size_t buffer_length) {
   int u8_size_needed = WideCharToMultiByte(CP_UTF8, 0, w_string.wstr_,
                                            static_cast<int>(w_string.size()),
-                                           buff, size_needed, NULL, NULL);
+                                           buff, buffer_length, NULL, NULL);
 
-  if (u8_size_needed != size_needed)
-    Fatal("Failed conversion to UTF-8 string");
+  if (u8_size_needed > (buffer_length - 1)) {
```

Comment:
Don'y you have this backwards?

The returned value will be between 0 and the provided length of the buffer.

The buffer you're given may be *exactly* the needed size, or it may be *larger*.

Reply:
If the buffer is exactly what we need then there's no room for the null termination. If the converted set is larger then the buffer supplied we should throw. So the u8_size_needed must be smaller than (buffer_length -1) for an ok value. If the buffers length is larger the check will succeed as the converted set of chars in smaller than the buffer. To me it looks like the check is ok.

Reply:
Ah, you're right.